queuelistener/listener.go

```diff
@@ -196,6 +201,15 @@ func Listen(o Options) (net.Listener, error) {
 		return nil, err
 	}
 
+	if o.ProxyProtocol {
+		log.Printf("Proxy protocol listener")
+		pl := &proxyproto.Listener{
+			Listener:          nl,
+			ReadHeaderTimeout: 10 * time.Second,
```

Comment:
IMO timeout should be taken from using `Options` struct, right? It shouldn't be hardcoded into the library itself, because you hide possible problems.
Look at the `maxConcurrency` and `maxQueueSize` functions inside this file and I'd go the same way as it's written here.